shared/js/background/components/resource-loader.js

```diff
@@ -0,0 +1,207 @@
+import Dexie from 'dexie'
+import { alarms } from 'webextension-polyfill'
+import bundledEtags from '../../../data/etags.json'
+import { createAlarm } from '../wrapper'
+/**
+ * @typedef {'tds' | 'surrogates' | 'config'} ResourceName
+ *
+ * @typedef {object} ResourceConfig
+ * @property {ResourceName} name
+ * @property {string} [remoteUrl]
+ * @property {string} [localUrl]
+ * @property {number} [updateIntervalMinutes]
+ * @property {'json'|'text'} [format]
+ *
+ * @typedef {(resourceName: ResourceName, etag: string, value: any) => void} OnUpdatedCallback
+ * @typedef {import('../settings.js')} Settings
+ */
+
+export default class ResourceLoader extends EventTarget {
+    /** @type {Dexie?} */
+    static dbc = null
+
+    /**
+     * @param {ResourceConfig} config
+     * @param {{
+     *  settings: Settings
+     * }} opts
+     */
+    constructor (config, { settings }) {
+        super()
+        this.settings = settings
+        this.name = config.name
+        this.remoteUrl = config.remoteUrl
+        this.localUrl = config.localUrl
+        this.updateIntervalMinutes = config.updateIntervalMinutes || 0
+        this.format = config.format || 'json'
+        this.data = null
+
+        if (!this.remoteUrl && !this.localUrl) {
+            throw new Error('invalid config: need a remote or local URL (or both)')
+        }
+
+        this.ready = this.checkForUpdates()
+
+        if (this.updateIntervalMinutes) {
+            const alarmName = `updateResource: ${this.name}`
+            createAlarm(alarmName, { periodInMinutes: this.updateIntervalMinutes })
+            alarms.onAlarm.addListener((alarmEvent) => {
```

Comment:
Let's use `browserWrapper.createAlarm` that already takes care of the checking logic. No point duplicating it here.